addons/implied-repos/common/src/test/java/org/commonjava/indy/implrepo/data/ValidReposStoreDataManagerTest.java

```diff
@@ -0,0 +1,85 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.commonjava.indy.implrepo.data;
+
+import org.commonjava.indy.audit.ChangeSummary;
+import org.commonjava.indy.data.StoreDataManager;
+import org.commonjava.indy.implrepo.fixture.TestValidReposStoreDataManager;
+import org.commonjava.indy.model.core.RemoteRepository;
+import org.commonjava.maven.galley.auth.MemoryPasswordManager;
+import org.commonjava.maven.galley.cache.FileCacheProvider;
+import org.commonjava.maven.galley.event.EventMetadata;
+import org.commonjava.maven.galley.event.NoOpFileEventManager;
+import org.commonjava.maven.galley.io.HashedLocationPathGenerator;
+import org.commonjava.maven.galley.io.NoOpTransferDecorator;
+import org.commonjava.maven.galley.maven.GalleyMaven;
+import org.commonjava.maven.galley.maven.GalleyMavenBuilder;
+import org.commonjava.maven.galley.transport.htcli.HttpClientTransport;
+import org.commonjava.maven.galley.transport.htcli.HttpImpl;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+
+import java.io.File;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+public class ValidReposStoreDataManagerTest
+{
+    @Rule
+    public TemporaryFolder temp = new TemporaryFolder();
+
+    private StoreDataManager storeManager;
+
+    private FileCacheProvider cache;
+
+    private GalleyMaven galley;
+
+    private File cacheDir;
+
+    private ChangeSummary summary;
+
+    @Before
+    public void setup()
+            throws Exception
+    {
+        cacheDir = temp.newFolder();
+        cache = new FileCacheProvider( cacheDir, new HashedLocationPathGenerator(), new NoOpFileEventManager(),
+                                       new NoOpTransferDecorator() );
+
+        galley = new GalleyMavenBuilder( cache ).withEnabledTransports(
+                new HttpClientTransport( new HttpImpl( new MemoryPasswordManager() ) ) ).build();
+
+        storeManager = new TestValidReposStoreDataManager( galley.getTransferManager() );
+
+        summary = new ChangeSummary( ChangeSummary.SYSTEM_USER, "test setup" );
+    }
+
+    @Test
+    public void testValidRepos()
+            throws Exception
+    {
+        RemoteRepository validRemote = new RemoteRepository( "test", "http://www.foo.com/" );
+        boolean valid = storeManager.storeArtifactStore( validRemote, summary, new EventMetadata() );
+        assertTrue( valid );
+
+        RemoteRepository inValidRemote = new RemoteRepository( "test", "http://this.is.not.a.real.repo/" );
+        boolean invalid = storeManager.storeArtifactStore( inValidRemote, summary, new EventMetadata() );
+        assertFalse( invalid );
+    }
+}
```

Comment:
If you need to check that URL validation is working for RemoteRepository instances, you can use ExpectationServer, as in: 

https://github.com/commonjava/indy/blob/3b5bde95d1d1a64fc38f141bb9194d1ec2e2380d/ftests/core/src/main/java/org/commonjava/indy/ftest/core/content/DownloadContentHasLengthHeaderTest.java